src/datum_blocktemplates.c

```diff
@@ -433,12 +433,29 @@ void *datum_gateway_template_thread(void *args) {
 		// fetch latest template
 		snprintf(gbt_req, sizeof(gbt_req), "{\"method\":\"getblocktemplate\",\"params\":[{\"rules\":[\"segwit\"]}],\"id\":%"PRIu64"}",(uint64_t)((uint64_t)time(NULL)<<(uint64_t)8)|(uint64_t)(i&255));
 		gbt = bitcoind_json_rpc_call(tcurl, &datum_config, gbt_req);
+		json_t *error_obj = json_object_get(gbt, "error");
 		
 		if (!gbt) {
 			datum_blocktemplates_error = "Could not fetch new template!";
 			DLOG_ERROR("Could not fetch new template from %s!", datum_config.bitcoind_rpcurl);
 			sleep(1);
 			continue;
+		} else if (json_is_object(error_obj)) {
```

Comment:
Would prefer to only check this if there's an issue with the normal path.

Reply:
No, it wouldn't make sense there. I mean before "Could not decode GBT result!" below:

https://github.com/OCEAN-xyz/datum_gateway/pull/132/files#diff-47011faa4cb93e4f0f17cd933d187da8517fe1fa21a0da2d49059eebacdf72c9L445